tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/Test_MagnetizedTovStar.cpp

```diff
@@ -34,46 +36,68 @@ static_assert(
 using TovCoordinates = RelativisticEuler::Solutions::TovCoordinates;
 
 void test_equality() {
-  const MagnetizedTovStar mag_tov_original{
-      1.28e-3, 100.0, 2.0, TovCoordinates::Schwarzschild, 2, 0.04, 2500.0};
-  const auto mag_tov = serialize_and_deserialize(mag_tov_original);
-  CHECK(mag_tov == MagnetizedTovStar(1.28e-3, 100.0, 2.0,
-                                     TovCoordinates::Schwarzschild, 2, 0.04,
-                                     2500.0));
-  CHECK(mag_tov != MagnetizedTovStar(2.28e-3, 100.0, 2.0,
-                                     TovCoordinates::Schwarzschild, 2, 0.04,
-                                     2500.0));
-  CHECK(mag_tov != MagnetizedTovStar(1.28e-3, 200.0, 2.0,
-                                     TovCoordinates::Schwarzschild, 2, 0.04,
-                                     2500.0));
-  CHECK(mag_tov != MagnetizedTovStar(1.28e-3, 100.0, 3.0,
-                                     TovCoordinates::Schwarzschild, 2, 0.04,
-                                     2500.0));
-  CHECK(mag_tov != MagnetizedTovStar(1.28e-3, 100.0, 2.0,
-                                     TovCoordinates::Isotropic, 2, 0.04,
-                                     2500.0));
-  CHECK(mag_tov != MagnetizedTovStar(1.28e-3, 100.0, 2.0,
-                                     TovCoordinates::Schwarzschild, 3, 0.04,
-                                     2500.0));
-  CHECK(mag_tov != MagnetizedTovStar(1.28e-3, 100.0, 2.0,
-                                     TovCoordinates::Schwarzschild, 2, 0.05,
-                                     2500.0));
-  CHECK(mag_tov != MagnetizedTovStar(1.28e-3, 100.0, 2.0,
-                                     TovCoordinates::Schwarzschild, 2, 0.04,
-                                     3500.0));
+  const MagnetizedTovStar mag_tov{
+      1.28e-3,
+      std::make_unique<EquationsOfState::PolytropicFluid<true>>(100.0, 2.0),
+      TovCoordinates::Schwarzschild,
+      2,
+      0.04,
+      2500.0};
+  // const auto mag_tov = serialize_and_deserialize(mag_tov_original);
+  CHECK(
+      mag_tov ==
+      MagnetizedTovStar(
+          1.28e-3,
+          std::make_unique<EquationsOfState::PolytropicFluid<true>>(100.0, 2.0),
+          TovCoordinates::Schwarzschild, 2, 0.04, 2500.0));
+  CHECK(
+      mag_tov !=
+      MagnetizedTovStar(
+          2.28e-3,
+          std::make_unique<EquationsOfState::PolytropicFluid<true>>(100.0, 2.0),
+          TovCoordinates::Schwarzschild, 2, 0.04, 2500.0));
+  // TODO : Need a comparison operator for arbitrary equations of state.
```

Comment:
This has been a bit of a contentious topic (can you/how should you compare abstract base classes). One way is to an `is_equal` virtual function that takes a `const EosBase&` argument, then have each derived EOS override the implementation, try to cast to itself, and if the cast works do the standard comparison. E.g.
```cpp
bool PolytropicFluid::is_equal(const Base& eos_base) const override {
  const auto& derived_ptr = dynamic_cast<const PolytropicFluid* const>(&eos_base);
  return derived_ptr != nullptr and *derived_ptr == *this;
}
```
I'm generally in favor of something like that since having some ability to do these types of comparisons is occasionally useful, but I'll bring it up at a spectre call (hopefully today)

Reply:
Any news on this?  I feel like I tried this but something seemed difficult, maybe it had to do with the templated base class and the feeling that as it stands such a function would pragmatically have to be templated and therefore couldn't be virtual.

Reply:
I checked with @kidder and he likes the idea. @wthrowe is having an `is_equal` function like that described above okay with you?

Reply:
To make sure, will this also entail adding in the `NewEosBase` class (or whatever we want to call it)?

Reply:
Seems fine.  I've seen something like this used before.  I think the only subtlety is that either all the classes that implement `is_equal` have to be final or the method should be private and the base class should have
```c++
friend bool operator==(const Base& a, const Base& b) {
  return a.is_equal(b) and b.is_equal(a);
}
```

Reply:
So will the idea to be something like 
```c++
namespace EquationsOfState {
Base : public PUP::able{
  ...
  friend bool operator==(const Base& a, const Base& b) {
    return a.is_equal(b) and b.is_equal(a);
}
}
template <bool IsRelativistic>
class EquationOfState<IsRelativistic, 1> : EquationOfStateBase {
  public:
  ...
```

Reply:
Sorry, I think just `a.is_equal(b)` should be enough. The checks should be symmetric :)

Reply:
`is_equal` should be symmetric, but if `a` and `b` have different dynamic types the two calls may be to different functions.  If the types are unrelated, this is still not a problem because the `dynamic_cast` expressions will both fail, but if one is derived from the other (and the more derived type overrides `is_equal`) the less derived `is_equal` can pass while the more derived one fails.  This is admittedly an obscure edge case.  (The more likely edge case is probably a base class with two derived classes, neither of which overrides `is_ready`, but symmetry isn't violated there.)

Perhaps a better solution is `typeid(a) == typeid(b) and a.is_equal(b)`.  That should avoid any subtleties.

Reply:
I like that suggestion :)

Reply:
I think the `TODO` comment can be removed now, right? Or at least once the `operator==` is used in the MTOV class :)